EphemeralElixir/EphemeralElixir
# Refactor and modularize code

#### What does this PR do?

Decouple code into their own modules:
  ~Request handlers for standard routes are in their own folder 
  ~Websocket and io connections have their own module
  ~Move activeUsers object into websocket.js

Refactor passport for better readability
Refactor server js to use config module for dbURI instead of hardcoding
#### Where should the reviewer start?

In the routes, websocket, and request handler files.
#### How should this be manually tested?

Go into the url and add /login. It should redirect to facebook for authentication and then redirect back to the site once authorized.
#### Any background context you want to provide?

Modularizing code this early will help lay out the ground work for when we begin writing out all of the socket emits and listeners on both the server-side and client-side. Smaller files in multiple modules will help us organize and locate problems should they arise.
#### What are the relevant tickets on Pivotal Tracker?

N/A
#### Provide screenshots (if appropriate).

N/A
